StormDBManager.py

```diff
@@ -0,0 +1,188 @@
+import logging
+from storm.database import Connection
+from storm.locals import *
+from storm.twisted.transact import Transactor, transact
+from twisted.internet.defer import DeferredLock
+
+from twisted.internet import reactor
+
+
+class StormDBManager:
+    """
+    The StormDBManager is a manager that runs queries using the Storm Framework.
+    These queries will be run on the Twisted thread-pool to ensure asynchronous, non-blocking behavior.
+    In the future, this database manager will be the basis to an ORM based approach.
+    """
+
+    def __init__(self, db_path):
+        """
+        Sets up the database and all necessary elements for the
+        database manager to function.
+        """
+        self._logger = logging.getLogger(self.__class__.__name__)
+
+        # Open or create the database
+        self._database = create_database(db_path)
+
+        # The transactor is required when you have methods decorated with the @transact decorator
+        # This field name must NOT be changed.
+        self.transactor = Transactor(reactor.getThreadPool())
+
+        # Create a DeferredLock that should be used by callers to schedule their call.
+        self.db_lock = DeferredLock()
+
+        self._version = 1
```

Comment:
On the current code, the initial database version is 0. Are you changing this for any particular reason?